presto-native-execution/presto_cpp/main/types/FunctionMetadata.cpp

```diff
@@ -0,0 +1,210 @@
+  }
+
+  protocol::RoutineCharacteristics routineCharacteristics;
+  routineCharacteristics.language =
+      std::make_shared<protocol::Language>(protocol::Language({"CPP"}));
+  routineCharacteristics.determinism =
+      std::make_shared<protocol::Determinism>(determinism);
+  routineCharacteristics.nullCallClause =
+      std::make_shared<protocol::NullCallClause>(nullCallClause);
+  return routineCharacteristics;
+}
+
+void updateFunctionMetadata(
+    const std::string& name,
+    const FunctionSignature& signature,
+    protocol::JsonBasedUdfFunctionMetadata& jsonMetadata) {
+  jsonMetadata.docString = name;
+  jsonMetadata.schema = kDefaultSchema;
+  jsonMetadata.outputType = signature.returnType().toString();
+
+  const std::vector<TypeSignature> types = signature.argumentTypes();
+  std::vector<std::string> paramTypes;
+  for (const auto& type : types) {
+    paramTypes.emplace_back(type.toString());
+  }
+  jsonMetadata.paramTypes = paramTypes;
+}
+
+const std::vector<protocol::JsonBasedUdfFunctionMetadata>
+buildAggregateJsonMetadata(
+    const std::string& name,
+    const std::vector<AggregateFunctionSignaturePtr>& signatures) {
+  std::vector<protocol::JsonBasedUdfFunctionMetadata> jsonMetadataList;
+  jsonMetadataList.reserve(signatures.size());
+  const protocol::FunctionKind kind = protocol::FunctionKind::AGGREGATE;
+
+  for (const auto& signature : signatures) {
+    protocol::JsonBasedUdfFunctionMetadata jsonMetadata;
+    jsonMetadata.functionKind = kind;
+    jsonMetadata.routineCharacteristics =
+        getRoutineCharacteristics(*signature, name, kind);
+    jsonMetadata.aggregateMetadata =
+        std::make_shared<protocol::AggregationFunctionMetadata>(
+            getAggregationFunctionMetadata(name, *signature));
+
+    updateFunctionMetadata(name, *signature, jsonMetadata);
+    jsonMetadataList.emplace_back(jsonMetadata);
+  }
+  return jsonMetadataList;
+}
+
+const std::vector<protocol::JsonBasedUdfFunctionMetadata>
+buildScalarJsonMetadata(
+    const std::string& name,
+    const std::vector<const FunctionSignature*>& signatures) {
+  std::vector<protocol::JsonBasedUdfFunctionMetadata> jsonMetadataList;
+  jsonMetadataList.reserve(signatures.size());
+  const protocol::FunctionKind kind = protocol::FunctionKind::SCALAR;
+
+  for (const auto& signature : signatures) {
+    protocol::JsonBasedUdfFunctionMetadata jsonMetadata;
+    jsonMetadata.functionKind = kind;
+    jsonMetadata.routineCharacteristics =
+        getRoutineCharacteristics(*signature, name, kind);
+
+    updateFunctionMetadata(name, *signature, jsonMetadata);
+    jsonMetadataList.emplace_back(jsonMetadata);
+  }
+  return jsonMetadataList;
+}
+
+const std::vector<protocol::JsonBasedUdfFunctionMetadata>
+buildWindowJsonMetadata(
+    const std::string& name,
+    const std::vector<FunctionSignaturePtr>& signatures) {
+  std::vector<protocol::JsonBasedUdfFunctionMetadata> jsonMetadataList;
```

Comment:
There isn't a need to use json in the names as that is implied. They can simply be metadataList, metadata etc.

Same comment is applicable for functions above as well.